docs/component_spec.md

```diff
@@ -0,0 +1,173 @@
+# Cytoxnet Component Specification
+The package is intended as an exploration and a toolbox for the prediction of
+cytotoxicity of compounds, but aims to leverage datasets of other targets
+for transfer learning.
+***
+***
+## package format
+```
+/cytoxnet
+|-/dataprep
+| |-io.py
+| |-featurize.py
+| |-dataprep.py
+| |-analyze.py
+|-/models
+| |-models.py
+| |-opt.py
+| |-analyze.py
+|-/tests
+```
+***
+
+## `cytoxnet/dataprep/`
+
+### `cytoxnet/dataprep/io.py`
+
+```
+load_file(datafile: str,
+          cols: str or list of str = None) -> pandas DataFrame
+```
+- __Use Case__: (1)
+- __Inputs__: File location string, columns to keep
+- __Outputs__: Dataframe of dataset on file
+
+```
+load_XXX(prepare: bool = False)
+```
+- __Use Case__: (1)
+- __Inputs__: Whether or not to featurize and prepare set using a default
+  pipeline
+- __Outputs__: Dataframe of package dataset
+
+### `cytoxnet/dataprep/featurize.py`
+
+```
+clean_dataframe(dataframe,
+                cols: str or list of str = None,
+                duplicates: str = 'drop',
+                nans: str = 'drop') -> pandas DataFrame
+```
+- __Use Case__: (1, 2)
+- __Inputs__: Dataframe, columns to clean, how to handle nans and duplicates.
+- __Outputs__: Dataframe
+
+```
+featurize(dataframe,
+          target_cols: list of str,
+          id_col: str = 'smiles',
+          featurizer: str = 'CircularFingerprint',
+          descriptor_cols: list of str = None) -> dataframe
+```
+- __Use Case__: (2)
+- __Inputs__: Dataframe, columns containing targets and features, type of
+  featurizer to use, additional descriptors to keep as features.
+- __Outputs__: Dataframe with X and y columns
+- *Temporary note*: call convert to catagorical on descriptor cols and target
+
+### `cytoxnet/dataprep/dataprep.py`
+
+```
+convert_to_catagorical(dataframe,
+                       cols: list of str)
+```
+- __Use Case__: (3)
+- __Inputs__: Dataframe, columns to consider for catagorical conversion
+- __Outputs__: Dataframe with considered columns that are non numeric converted.
+
+```
+convert_to_dataset(dataframe,
+                   X_col: str = 'X',
+                   y_col: str = 'y',
+                   w_col: str = None)
```

Comment:
```suggestion
convert_to_dataset(dataframe,
                   X_col: str = 'X',
                   y_col: str = 'y',
                   w_col: str = None,
                   id_col: str = None)
```
Should probably add argument for id column since deepchem function take an `ids` argument.